contracts/Minter.sol

```diff
@@ -0,0 +1,116 @@
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/IERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/token/ERC20/utils/SafeERC20Upgradeable.sol";
+import "@openzeppelin/contracts-upgradeable/utils/cryptography/MerkleProofUpgradeable.sol";
+
+interface Daomon721a {
+    function mint(address to, uint256 amount) external;
+}
+
+contract Minter is OwnableUpgradeable {
+    using SafeERC20Upgradeable for IERC20Upgradeable;
+
+    error NoPurchasableTokens();
+    error ZeroAmountNotAllowed();
+    error NotEnoughMintableTokensForThisUser();
+    error ReceiverNotWhitelisted();
+
+    // User variables
+    uint256 public numOfPurchasedTokens;
+    mapping(address => uint256) public numOfMintableTokensPerAddress;
+
+    // Admin variables
+    uint256 public deploymentTimeStamp;
+    uint256 public priceInPMON;
+    uint256 public dailySupply;
+    address public pmonAddress;
+    address public pmonReceiverAddress;
+    address public daomon721aAddress;
+    bytes32 public merkleRoot;
+
+    function initialize(
+        address _pmonReceiverAddress,
+        address _daomon721aAddress,
+        bytes32 _merkleRoot
+    ) public initializer {
+        deploymentTimeStamp = block.timestamp;
+        dailySupply = 10;
+        priceInPMON = 10e18;
```

Comment:
Why this low init value? I would go for a higher value in the init phase to avoid any security issues.

Reply:
To make it more exclusive 👀